File: BubbleSort/C++/bubbleSort.cpp


#include <iostream>
using namespace std;

void bubbleSort(int arr[], int size) {
    bool isSorted;
    for (int i = 0; i < size; i++) {
        isSorted=true;
        for (int j = 1; j < size - i; j++) {
            if (arr[j] < arr[j - 1]) {
                swap(arr[j], arr[j - 1]);
                isSorted = false;
            }
        }
        if (isSorted)return;
    }
}

File: BubbleSort/JS/bubbleSort.js
const swap = (arr, index1, index2) => {
    let temp = arr[index1];
    arr[index1]=arr[index2];
    arr[index2]=temp;
}

const bubbleSort = arr => {
    let isSorted;
    for (let i=0; i<arr.length; i++){
        isSorted=true;
        for (let j=1; j<arr.length-i; j++){
            if (arr[j]<arr[j-1]){
                swap(arr, j, j-1);
                isSorted=false;
            }
        }
        if (isSorted) return;
    }
}

/*let arr = [1,8,9,7,3,2,1];
bubbleSort(arr);
console.log(arr);*/
